### PR TITLE
CI: Lint checks failures are reporting success

### PR DESCRIPTION
**Pull request by peterbell10**
#### What does this implement/fix?
If you run lint checks manually, you can see that several lint checks are failing on master but CI was succeeding. This is because by default a bash script returns the exit code of the last command, ignoring intermediate commands. The fix is putting `set -e` at the start of the bash script, which fails immediately if any command has non-zero exit code.

### CODE REVIEW

**Comment by peterbell10** [comment]
And indeed the lint CI is now failing with many violations
https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=12972&view=logs&j=fdd89725-f657-5341-7d77-3c5c44c4f5b6&t=325fc03e-e0b9-5339-adaf-099a49a14600

**Comment by larsoner** [comment]
Agreed this is useful, feel free to fix those violations and +1 for merge after that

File: azure-pipelines.yml [hunk 1]
```diff
@@ -193,6 +194,7 @@ stages:
         git submodule update --init
       displayName: 'Fetch submodules'
     - script: |
+            set -e
```
Comment (tupui) [inline azure-pipelines.yml:197]: If we do this, why not going all the way to `set -euxo pipefail`

https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

**Review by tupui — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -169,11 +169,14 @@ stages:
     - script: >-
         python -m pip install
         pycodestyle==2.5.0
-        pyflakes==2.1.1
+        pyflakes==2.3.1
       displayName: 'Install tools'
       failOnStderr: false
     - script: |
-        PYFLAKES_NODOCTEST=1 pyflakes scipy benchmarks/benchmarks 2>&1 | grep -E -v 'unable to detect undefined names|assigned to but never used|imported but unused|redefinition of unused|may be undefined, or defined from star imports|syntax error in type comment .ignore.import.' > test.out; cat test.out; test \! -s test.out
+        set -e
```
Comment (tupui) [inline azure-pipelines.yml:176]:
```suggestion
        set -euxo pipefail
```

**Review by tupui — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (peterbell10) [reply azure-pipelines.yml]: The `set -x` part writes to `stderr` which makes the CI run fail. We can also allow `stderr` output, but that risks warnings being missed.

Comment (tupui) [reply azure-pipelines.yml]: Sure the -x is not really needed as already done by default by the CI.

**Review by ilayn — APPROVED** [review]

**Comment by tupui** [comment]
Since we have another approval, let's get this in. Thanks @peterbell10.

**Pull request merged**